lib/diagnostic.rb

```diff
@@ -47,5 +47,13 @@ def transform(words)
   # space and return the result.
   def read_file(filename)
     # Your code goes here
+    line_num=0
+    concat_lines = ''
+    text=File.open(filename).read
+    text.gsub!(/\r\n?/, " ")
+    text.each_line do |line|
+      concat_lines += line
+    end
+    concat_lines
   end
```

Comment:
So close! it seems that your code did not remove the `\n`s in the text. But you're so close!